### DISCUSSION TITLE
Related resource index view.

### DISCUSSION THREAD

**Discussion by philipbaginski** [laravel/nova-issues#5171] [Q&A] [answered]
Hi everyone!

What is my question: I have a resource X related to Y by belongs to many relation. On detail view of resource Y I can see X resource index table with its fields, and in addition pivot fields. That is fine, but resource X has many fields on index view I don't want to display when index view is displayed via relationship. Right now I have two options: Create extra resource just for displaying index view of X resource via relationship. Second, in resource X use fieldsForIndex and in addition use

​```
->hideFromIndex(function (ResourceIndexRequest $request) {
                return $request->viaRelationship();
            }),
​```
We have a set of fields for many cases like fieldsForIndex, fieldsForDetail, etc., but there is no fieldsForRelationship.

What do you think about it?

**Comment by crynobone** [comment, marked as answer]
You can always create field macro for reusable/repeating filters.

**Reply by philipbaginski** [reply]
Are you sure it's a good idea for related resource index view?

**Reply by crynobone** [reply]
I don't have the use case to code it but `$request->viaRelationship()` and `$request->viaManyToMany()` is reliable methods to determine such usage.

**Comment by philipbaginski** [comment]
So, my case:

Resource Trainer, index view fields.

​```

    /**
     * Get the fields displayed by the resource - index view.
     *
     * @param  \Laravel\Nova\Http\Requests\NovaRequest  $request
     * @return array
     */
    public function fieldsForIndex(NovaRequest $request)
    {
        return [
            Text::make(__('Trainer'), 'name')
                ->sortable(),
            BelongsTo::make(__('Country'), 'country', Country::class)
                ->sortable()
                ->viewable(false)
                ->hideFromIndex(function (ResourceIndexRequest $request) {
                    return $request->viaRelationship();
                }),
            Text::make(__('Phone'), fn () => $this->trainerProfile->phone ?? null)
                 ->hideFromIndex(function (ResourceIndexRequest $request) {
                     return $request->viaRelationship();
                 }),
            Email::make(__('Email'), 'email')
                ->sortable()
                ->hideFromIndex(function (ResourceIndexRequest $request) {
                    return $request->viaRelationship();
                }),
            Text::make('ID', 'id')
                ->sortable()
                ->hideFromIndex(function (ResourceIndexRequest $request) {
                    return $request->viaRelationship();
                }),
            BelongsToMany::make(__('Players'), 'players', Player::class)
                ->fields(function () {
                    return [
                        Date::make(__('From'), 'from_date'),
                        Date::make(__('Until'), 'to_date'),
                        Textarea::make(__('Notes'), 'notes')
                            ->rows(2),
                        Boolean::make(__('Is Active'), 'is_active')
                            ];
                        })
        ];
    }

​```   

, and via relationship on the index view I want to see only trainer name. Right now to every field I don't want to see on the index via relationship I have do add

​```
->hideFromIndex(function (ResourceIndexRequest $request) {
                    return $request->viaRelationship();
                }),
​```

to hide them. So, my question is how to make fieldsForViaRelationship? :-)

**Reply by crynobone** [reply]
Field + Macro: https://crynobone.com/add-field-default-only-on-create-on-laravel-nova/

**Reply by philipbaginski** [reply]
Thanx. Will try.

**Comment by cweiske** [comment]
Related issue: https://github.com/laravel/nova-issues/issues/1697
